source/server/overload_manager_impl.h

```diff
@@ -52,6 +52,52 @@ class OverloadAction {
   Stats::Gauge& scale_percent_gauge_;
 };
 
+// Simple table that converts strings into Symbol instances. Symbols are guaranteed to start at 0
+// and be indexed sequentially.
+class NamedOverloadActionSymbolTable {
+public:
+  class Symbol {
+  public:
+    // Allow copy construction everywhere.
+    Symbol(const Symbol&) = default;
+
+    // Returns the index of the symbol in the table.
+    size_t index() const { return index_; }
+
+    // Support use as a map key.
+    bool operator==(const Symbol other) { return other.index_ == index_; }
+
+    // Support absl::Hash.
+    template <typename H>
+    friend H AbslHashValue(H h, const Symbol& s) { // NOLINT(readability-identifier-naming)
+      return H::combine(std::move(h), s.index_);
+    }
+
+  private:
+    friend class NamedOverloadActionSymbolTable;
+    // Only the symbol table class can create Symbol instances from indices.
+    explicit Symbol(size_t index) : index_(index) {}
+
+    size_t index_;
+  };
+
+  // Finds an existing or adds a new entry for the given name.
+  Symbol get(absl::string_view name);
+
+  // Returns the symbol for the name if there is one, otherwise nullopt.
+  absl::optional<Symbol> lookup(absl::string_view string) const;
+
+  // Translates a symbol back into a name.
+  const std::string& name(Symbol symbol) const;
+
+  // Returns the number of symbols in the table. All symbols are guaranteed to have an index less
+  // than size().
+  size_t size() const { return table_.size(); }
+
+  absl::flat_hash_map<std::string, size_t> table_;
```

Comment:
Should these be private?

Reply:
Whoops, thanks. That's kind of the whole point of this class.